client/my-sites/plan-features-2023-grid/grid-context.tsx

```diff
@@ -0,0 +1,21 @@
+import { createContext, useContext } from '@wordpress/element';
+import type { Intent } from 'calypso/my-sites/plans-features-main/hooks/use-plan-types-with-intent';
+
+interface PlansGridContext {
+	intent: Intent;
```

Comment:
Ideally, the actual grid package would not need to know anything about intent. So eventually deprecating this will take things a step further (it will mean feeding the grid with the respective plan features and the highlight labels for plans). 

Context can stay though and can be adopted to cover all the props passed into the grid when consumed/imported.